### ISSUE TITLE
Waiting for server

### ISSUE THREAD

**Issue by raulimann** [NillerMedDild/Enigmatica6#2354]
**Modpack version**
0.4.14

**Describe the bug**
In singleplayer the internal server starts to lag when exploring the world, which results in an endless "waiting for server". World interactions are no longer possible and the world stops generating. The client side continues to work without problems.

**Logs**
[latest.log](https://github.com/NillerMedDild/Enigmatica6/files/6602236/latest.log)

**Screenshots**
![grafik](https://user-images.githubusercontent.com/56782788/120885463-59d33d80-c5e9-11eb-840d-6aa738cd4cb0.png)

**Enviroment**
8700k, 1080ti, 32GB; tested with Java 12.0.2.0 and 1.8.0.51; allocated between 8 and 12 GB memory

**Comment by landretk** [comment]
This happens in multiplayer too but I haven't had a good way to reproduce it. I suspect it's due to DungeonCrawl because it procedurally generates dungeons. If you can reliably reproduce it I would start pulling out mods 1 by 1 that generate features, like Tetra, waystones, etc. Could be an issue in Terraforged itself which would be unfortunate but most likely it's one of these dungeon mods or something getting stuck in a loop while generating a feature.

You should be able to mitigate it by reducing your view distance and/or pregenerating some chunks with https://www.curseforge.com/minecraft/mc-mods/chunkpregenerator which is included in the pack, if you are only interested in fixing it and not in debugging.

A debug.log from a run where this happened would also be helpful, since forge and terraforged are supposed to log cascading world gen problems

**Comment by NillerMedDild** [comment]
Finding a way to reproduce this would indeed help a ton in this case.
OP do you have a debug.log from when the issue happened? I know the CurseForge Launcher turns them off sadly.

**Comment by raulimann** [comment]
I had to import the modpack into MultiMC to generate a debug.log, but I managed it to recreate the error there too.
[debug.log](https://github.com/NillerMedDild/Enigmatica6/files/6603145/debug.log)

**Comment by NillerMedDild** [comment]
Awesome, thanks for doing that! Do you know approx. when it happened?

**Comment by raulimann** [comment]
Sure, sometime between 21:37:19 and 21:39:50

**Comment by NillerMedDild** [comment]
Alright, so you encountered 3 Better Strongholds, and a Dungeon Crawl dungeon in that period. I'll try to see if I can reproduce it by going to either.

Snippet of the log:
```
[05Jun2021 21:37:19.244] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT] Activated Waystone: Mrokcrader
[05Jun2021 21:37:31.375] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:drawer_key on screen IngameMenuScreen.
[05Jun2021 21:37:32.616] [Render thread/INFO] [Astral Sorcery/]: Client cache cleared!
[05Jun2021 21:37:32.638] [Render thread/DEBUG] [com.simibubi.create.Create/]: Removed Redstone Network Space for minecraft:overworld
[05Jun2021 21:37:32.639] [Render thread/DEBUG] [com.simibubi.create.Create/]: Removed Kinetic Network Space for minecraft:overworld
[05Jun2021 21:37:41.376] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:ee_ores on screen DirtMessageScreen.
[05Jun2021 21:37:44.612] [Worker-Main-33/DEBUG] [betterstrongholds/]: Better Stronghold at -3545 29 4343
[05Jun2021 21:37:49.670] [Worker-Main-34/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@21309783' to registry
[05Jun2021 21:37:51.377] [Render thread/DEBUG] [Tips/]: Displaying tip tips:baby_panda_slimeball on screen DirtMessageScreen.
[05Jun2021 21:37:53.758] [Worker-Main-33/DEBUG] [betterstrongholds/]: Better Stronghold at 8567 29 -6441
[05Jun2021 21:37:54.000] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Creating the layout for a dungeon at (6753 | 47 | 5297) with 5 layers. Theme: 0 Sub-Theme: 0
[05Jun2021 21:37:54.053] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished basic generation of layer 0: Generated 3/3 nodes and 4/4 rooms.
[05Jun2021 21:37:54.053] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished generation of layer 0
[05Jun2021 21:37:54.053] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished basic generation of layer 1: Generated 3/4 nodes and 5/5 rooms.
[05Jun2021 21:37:54.055] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished generation of layer 1
[05Jun2021 21:37:54.056] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished basic generation of layer 2: Generated 5/5 nodes and 7/7 rooms.
[05Jun2021 21:37:54.056] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished generation of layer 2
[05Jun2021 21:37:54.056] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished basic generation of layer 3: Generated 5/6 nodes and 8/8 rooms.
[05Jun2021 21:37:54.056] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished generation of layer 3
[05Jun2021 21:37:54.056] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished basic generation of layer 4: Generated 6/7 nodes and 10/10 rooms.
[05Jun2021 21:37:54.056] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Finished generation of layer 4
[05Jun2021 21:37:54.081] [Worker-Main-33/DEBUG] [Dungeon Crawl/]: Created the dungeon layout for [426, 335] with a total of 308 pieces.
[05Jun2021 21:38:01.386] [Render thread/DEBUG] [Tips/]: Displaying tip tips:animal_growing on screen DirtMessageScreen.
[05Jun2021 21:38:08.566] [Worker-Main-31/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:38:11.386] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:mob_farms on screen DirtMessageScreen.
[05Jun2021 21:38:21.387] [Render thread/DEBUG] [Tips/]: Displaying tip tips:piglin_gold on screen DirtMessageScreen.
[05Jun2021 21:38:28.574] [Worker-Main-24/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@21309783' to registry
[05Jun2021 21:38:31.641] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:back_home on screen DirtMessageScreen.
[05Jun2021 21:38:37.449] [Worker-Main-24/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@21309783' to registry
[05Jun2021 21:38:41.642] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:horologium on screen DirtMessageScreen.
[05Jun2021 21:38:48.361] [Worker-Main-24/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:38:51.644] [Render thread/DEBUG] [Tips/]: Displaying tip tips:creeper_cat on screen DirtMessageScreen.
[05Jun2021 21:38:58.637] [Worker-Main-24/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:39:01.645] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:cow_level on screen DirtMessageScreen.
[05Jun2021 21:39:05.056] [Worker-Main-32/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:39:05.198] [Worker-Main-35/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@21309783' to registry
[05Jun2021 21:39:11.664] [Render thread/DEBUG] [Tips/]: Displaying tip tips:water_grass on screen DirtMessageScreen.
[05Jun2021 21:39:22.208] [Render thread/DEBUG] [Tips/]: Displaying tip tips:boat_passenger on screen DirtMessageScreen.
[05Jun2021 21:39:24.632] [Worker-Main-32/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:39:32.210] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:modded_for_dummies on screen DirtMessageScreen.
[05Jun2021 21:39:38.471] [Worker-Main-34/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:39:40.294] [Worker-Main-24/DEBUG] [betterstrongholds/]: Better Stronghold at -5383 29 9721
[05Jun2021 21:39:42.211] [Render thread/DEBUG] [Tips/]: Displaying tip enigmatica:prettypipes_facade on screen DirtMessageScreen.
[05Jun2021 21:39:51.433] [Worker-Main-24/ERROR] [net.minecraft.util.registry.SimpleRegistry/]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@26fdfa64' to registry
[05Jun2021 21:39:51.723] [Server thread/INFO] [net.minecraft.network.play.ServerPlayNetHandler/]: raulimann lost connection: Disconnected
[05Jun2021 21:39:51.724] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: raulimann left the game
```

**Comment by landretk** [comment]
I am highly suspicious of Dungeon Crawl because I had an issue with occasional hanging worldgen shortly after adding it to a pack I was working on myself. I took it out to be safe.

**Comment by NillerMedDild** [comment]
Oddly enough, the issue happened after I had found 1 Dungeon Crawl dungeon and 3 Better Strongholds. Might be a coincidence though.

My latest.log if anyone else is listening in 😄 
[latest.log](https://github.com/NillerMedDild/Enigmatica6/files/6603170/latest.log)

**Comment by landretk** [comment]
Hmm, looking at the source of both though, DC does not appear to use JigsawManager, but Better Strongholds does

**Comment by NillerMedDild** [comment]
So I guess the next step is reproducing it without Better Strongholds. I need sleep first though :P

**Comment by landretk** [comment]
May be an interaction between Terraforged and it https://github.com/yungnickyoung/YUNGs-Better-Strongholds/issues/9

**Comment by NillerMedDild** [comment]
So I ran VisualVM and found out that the `locate betterstronghold:stronghold` command itself created a deadlock >_<

**Comment by NillerMedDild** [comment]
I managed to find the deadlock itself and run VisualVM:
[snapshot.zip](https://github.com/NillerMedDild/Enigmatica6/files/6604896/snapshot.zip)

**Comment by NillerMedDild** [comment]
The TF dev suggested that it could be our TF performance configs messing with things - OP would you mind deleting the config file `config/terraforged/performance_internal.conf` and see if you can still reproduce the issue?

**Comment by raulimann** [comment]
Nope, still the same issue :[

![2021-06-07_17 11 52](https://user-images.githubusercontent.com/56782788/121042189-b84c1780-c7b3-11eb-940c-345174c5bad1.png)

**Comment by NillerMedDild** [comment]
Thanks! 👍

**Comment by NillerMedDild** [comment]
I cannot reproduce this without BetterStrongholds installed :/ we're gonna have to remove it.

**Comment by WhiteFire-Sondergaard** [comment]
I am still seeing this issue as of 0.5.11 (installed via ATLancher)

`[18/09/2021 18:40:59 PM] [18:40:59] [Worker-Main213/ERROR] [minecraft/SimpleRegistry]: Adding duplicate value 'net.minecraft.world.gen.feature.jigsaw.JigsawPattern@717286d0' to registry`

Happens after some random amount of chunk generation making exploring in single player problematic at best.

**Comment by WhiteFire-Sondergaard** [comment]
After digging around the error messages a bit more they seem to relate to various Towers of the Wild patterns being broken, error messages such as "Emty or none existent pool: towers_of_the_wild:ocean_bottom" seem consistent with this issue off that mod:

https://githubmemory.com/repo/Idrae/Towers_Of_The_Wild-1.16.3/issues

See the Blame report listed there.

**Comment by NillerMedDild** [comment]
That error message has been around for ages, and has nothing to do with this closed issue :)

**Comment by yuesha-yc** [comment]
Tower of the wild issue is fixed in this branch: https://github.com/TeamMoegMC/TowersOfTheWildReloaded
I think E6 already included it in latest beta.

**Comment by Pommynouche** [comment]
So I was wondering if the issue got fixed ? I still get it from time to time, is it only me ?

**Comment by Kanzaji** [comment]
It is very old, and appears to be fixed completely at this point 😅
If you have some other problems however, you can ask for help in support channel on Enigmatica Discord :)

PS: If your server / singleplayer world doesn't freeze up completely, just gets "waiting for server" for a bit sometimes, then this is normal ^^" Your world is starting to just get laggy >.>

**Comment by miner-beep** [comment]
Hi my friends I'm experiencing the same problem but much worse it happens to me after 5 minutes of creating a default world and happens as soon as i creat e a biome'o planty world . I'm using better minecraft plus v38 and i added 100 mods myself so it now has 406 mods i was not experiencing this problem at first . I deleted all of those mods that you mentioned like terra dungeon crawl better stronghold and wild towers can you please help me

**Comment by LordWabbit** [comment]
FYI - I had the same issue, started updating mods that seemed like they were the likely culprits (and any that a long google search suggested) and then I decided to try run the server portion on it's own to help clean up the logs and try find a useful error message the server complained about the architectury mod when trying to start up.  I went and checked and I had TWO of them in my mod folder.  Removed both, and installed the latest and the problem went away.  

This is a generic error that can be caused by lots of things, this may not be your problem, but I just thought I would mention it in case it helps someone in the future.
